src/snowflake/cli/api/cli_global_context.py

```diff
@@ -17,14 +17,16 @@
 import re
 import warnings
 from pathlib import Path
-from typing import Callable, Optional
+from typing import TYPE_CHECKING, Callable, Optional
 
 from snowflake.cli.api.exceptions import InvalidSchemaError
 from snowflake.cli.api.output.formats import OutputFormat
-from snowflake.cli.api.project.schemas.project_definition import ProjectDefinition
 from snowflake.connector import SnowflakeConnection
 from snowflake.connector.compat import IS_WINDOWS
 
+if TYPE_CHECKING:
+    from snowflake.cli.api.project.schemas.project_definition import ProjectDefinition
```

Comment:
Importing `ProjectDefinition` from this module creates a circular dependency if any of the downstream modules try to `cli_console.warning()` (or step/message/etc).

Up until now we managed to avoid the cycle because the project definition did not have references to the business logic part of the code. For Workspaces we need to map each schema to its entity (`v2_entity_model_to_entity_map`).

This is the import cycle:
```
snowflake.cli.api.cli_global_context ->
snowflake.cli.api.project.schemas.project_definition ->
snowflake.cli.api.project.schemas.entities.entities ->
snowflake.cli.api.entities.application_package_entity ->
snowflake.cli._plugins.nativeapp.codegen.compiler ->
snowflake.cli._plugins.nativeapp.codegen.setup.native_app_setup_processor ->
snowflake.cli._plugins.stage.diff ->
snowflake.cli.api.console ->
snowflake.cli.api.console.console ->
snowflake.cli.api.console.abc ->
snowflake.cli.api.cli_global_context
```

Reply:
it's a bit surgical to do it this way, but it will do.